Library/Homebrew/system_config.rb

```diff
@@ -85,6 +85,19 @@ def describe_ruby
       end
     end
 
+    def describe_system_ruby
+      s = ""
+      case RUBY_VERSION
+      when /^1\.[89]/, /^2\.0/
```

Comment:
Look at e.g. `dump_generic_verbose_config` or `grep` for `def generic_` and you'll see how they are used. In short: define a generic method in the main class, call it from the `os` class.